lib/ide/src/view/layout.rs

```diff
@@ -136,43 +105,39 @@ impl ViewLayoutData {
         self.text_editor.set_padding(padding);
         self.text_editor.set_size(size);
         self.text_editor.set_position(position);
-        self.text_editor.update();
     }
 }
 
 impl ViewLayout {
     /// Creates a new ViewLayout with a single TextEditor.
-    pub fn default(world:&World) -> Self {
-        let text_editor  = TextEditor::new(&world);
-        let key_listener = None;
+    pub fn new
+    ( key_actions          : &mut KeyboardActions
```

Comment:
either use `keyboard_actions` or `kb_actions`. "key" is not a valid abbreviation from "keyboard" :)